internal/worker/jobworker/jobworker.go

```diff
@@ -0,0 +1,186 @@
+package jobworker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/joshuarubin/teleport-job-worker/internal/config"
+	"github.com/joshuarubin/teleport-job-worker/internal/job"
+	"github.com/joshuarubin/teleport-job-worker/internal/user"
+	"github.com/joshuarubin/teleport-job-worker/internal/worker"
+)
+
+// ReexecCommand contains the necessary configuration for the JobWorker to be
+// able to call exec.Command() and expect that the current binary will be
+// reexecuted, the JobWorker will be reinstantiated, and
+// JobWorker.StartJobChild() will be called with the remaining arguments passed
+// in as command and args
+type ReexecCommand struct {
+	Command string   // often "/proc/self/exe"
+	Args    []string // usually a command that puts this binary in a different "mode", e.g. "child" or "helper"
+	Env     []string // additional env variables, in the form of "key=value", to be added to the current os.Environ()
+}
+
+type cgroup struct {
+	cfg        *config.Config
+	createOnce sync.Once //nolint:unused,nolintlint
+	createErr  error     //nolint:unused,nolintlint
+	name       string    //nolint:unused,nolintlint
+}
+
+// JobWorker is an implementation of the Worker interface
+type JobWorker struct {
+	cfg    *config.Config
+	reexec *ReexecCommand
+
+	cgroup *cgroup
+
+	mu   sync.RWMutex
+	jobs map[job.ID]*job.Job
+}
+
+// ensure JobWorker implements the Worker interface
+var _ worker.Worker = (*JobWorker)(nil)
```

Comment:
Could we ditch the interface and expose JobWorker directly?

The interface is useful to express the API in the design phase, but I don't think the API actually calls for it.

- https://go.dev/wiki/CodeReviewComments#interfaces

Reply:
Fine with me!